mayiPalacios/xtwitter
# M3 practice lab 2

**Overview**


This pull request adds route definitions and routes specs for various features as outlined in the lab instructions. No actual logic has been implemented in the controllers as this is part of Practice Lab 3. The purpose of this pull request is to set up the necessary routes and routes specs.


**Route Definitions**


- Created routes for the following features:

  - Create tweet

  - Update tweet

  - Like Tweet

  - Unlike Tweet

  - Retweet

  - Quote

  - Reply

  - Bookmark

  - User Tweets (with pagination)

  - User Tweets & Replies (with pagination)

  - Tweet Stats


- All routes are defined in `config/routes.rb` and are visible via `bin/rails routes` or `[http://localhost:3000/rails/info/routes`](http://localhost:3000/rails/info/routes).


**Route Specs**


- Included routes specs for each route in the `spec/routes` directory.



**Testing**


- Ensure that routes can be seen using `bin/rails routes` or by visiting `[http://localhost:3000/rails/info/routes`](http://localhost:3000/rails/info/routes).

- Verify that route specs are correctly defined and pass by running `rspec` or the testing command specified in the project.


**Note**


- No actual implementation logic is included in controllers, as instructed
